src/components/textfieldSlider/js/TextfieldSlider.js

```diff
@@ -66,123 +62,117 @@ var fluid_2_0 = fluid_2_0 || {};
 
     fluid.defaults("fluid.textfieldSlider.textfield", {
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.stringToNumber",
+                input: "{that}.model.stringValue"
+            }
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            },
+            // When the input field receives a invalid input such as a string value, ignore it and populate the field with the previous value
+            stringValue: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{that}.model.value"]
+            }
+        },
         listeners: {
-            onCreate: {
-                listener: "fluid.textfieldSlider.textfield.init",
-                args: "{that}"
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
             }
         },
-        range: {} // should be used to specify the min, max range e.g. {min: 0, max: 100}
-    });
-
-    fluid.textfieldSlider.validateValue = function (model, range, changeRequest) {
-        var oldValue = model.value;
-        var newValue = changeRequest.value;
-
-        if (!isNaN(parseInt(newValue, 10))) {
-            if (newValue < range.min) {
-                newValue = range.min;
-            } else if (newValue > range.max) {
-                newValue = range.max;
+        invokers: {
+            setModel: {
+                changePath: "stringValue",
+                value: "{arguments}.0.target.value"
             }
-            changeRequest.value = Number(newValue);
-        } else {
-            changeRequest.value = oldValue;
         }
-    };
-
-    fluid.textfieldSlider.textfield.init = function (that) {
-        that.applier.guards.addListener({path: "value", transactional: true}, function (model, changeRequest) {
-            fluid.textfieldSlider.validateValue(model, that.options.range, changeRequest);
-        });
-
-        that.container.change(function (source) {
-            that.applier.requestChange("value", source.target.value);
-        });
-    };
+    });
 
     fluid.defaults("fluid.textfieldSlider.slider", {
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
         selectors: {
             thumb: ".ui-slider-handle"
         },
-        events: {
-            modelChanged: null
+        members: {
+            slider: {
```

Comment:
Just convert this into a genuine Infusion component named fluid.slider and we can eliminate the oddity of the member + expander and the strange function which just dispatches to its function argument. This can eventually become elevated to a standard wrapper along the same lines as the existing "fluid.tooltip" etc.